### ISSUE TITLE
All modules are linted in watch mode after changing only a single file

### ISSUE THREAD

**Issue by decademoon** [webpack-contrib/eslint-webpack-plugin#75]
- Operating System: macOS 11.1
- Node Version: 15.7.0
- NPM Version: yarn berry PnP
- webpack Version: 4.46.0
- eslint-webpack-plugin Version: 2.5.0

### Expected Behavior

Only lint modules that were rebuilt after changing a file.

### Actual Behavior

Every module is linted after changing a file.

### Code

```js
new ESLintPlugin({
  extensions: ['js', 'vue'],
})
```

### How Do We Reproduce?

It seems there was a recent change to which hooks were used to lint the files. Before `succeedModule` was used which was only called with the rebuilt modules after a file change, but now `finishModules` is used which passes every module in the compilation, so all files get linted.

**Comment by ricardogobbosouza** [comment]
Solving it now. =)

**Comment by WilkinWendy** [comment]
My problem is a little different. In my project, there remains a lot of errors & warnings. I only what to lint what is saved by current user on his IDE.
Now, when the project starts, all files are linted..about 100+

**Comment by ricardogobbosouza** [comment]
@WilkinWendy for this you can enable `lintDirtyModulesOnly`

**Comment by WilkinWendy** [comment]
> @WilkinWendy for this you can enable `lintDirtyModulesOnly`

I’ve tried it，but not work.

**Comment by WilkinWendy** [comment]
> > @WilkinWendy for this you can enable `lintDirtyModulesOnly`
> 
> I’ve tried it，but not work.

` {
                exclude: 'node_modules',
                extensions: ['.js', '.jsx', '.vue', '.ts', '.tsx'],
                quiet: true,
                cache: true,
                fix: true,
                lintDirtyModulesOnly: true,
                formatter: 'codeframe'
            }`

**Comment by WilkinWendy** [comment]
> > @WilkinWendy for this you can enable `lintDirtyModulesOnly`
> 
> I’ve tried it，but not work.

In my project, the compilation seems to be triggered in a dead loop, when `fix:true`

**Comment by WilkinWendy** [comment]
> > > @WilkinWendy for this you can enable `lintDirtyModulesOnly`
> > 
> > 
> > I’ve tried it，but not work.
> 
> In my project, the compilation seems to be triggered in a dead loop, when `fix:true`

I‘ve worked out reason of this problem.
If a file is unfixable,which means the output of the fixed file is the same as the source file,
eg: `no-console` rule is unfixable.
the outputFixes function will be executed in a dead loop.
